modules/programs/fish.nix

```diff
@@ -12,6 +12,23 @@ let
     mapAttrsFlatten (k: v: "alias ${k} '${v}'") cfg.shellAliases
   );
 
+  babelfish = pkgs.callPackage ../../pkgs/shells/fish/plugins/babelfish.nix { };
+
+  envShellInit = pkgs.writeText "shellInit" cfge.shellInit;
+
+  envLoginShellInit = pkgs.writeText "loginShellInit" cfge.loginShellInit;
+
+  envInteractiveShellInit = pkgs.writeText "interactiveShellInit" cfge.interactiveShellInit;
+
+  sourceEnv = useTranslated: path:
+  if useTranslated then
+    "source ${path}"
+  else
+    ''
+      set fish_function_path ${pkgs.fishPlugins.foreign-env}/share/fish-foreign-env/functions $fish_function_path
+      fenv source ${path} > /dev/null
```

Comment:
Do we still need `fenv` there as we are using `babelfish`?

Reply:
We don't. In the version I've done for nixOS, this requirement is removed. I'll go ahead and port it back to nix-darwin in this branch.